### PR TITLE
fix(ocient): convert pyocient GIS data types to geoJSON

### PR DESCRIPTION
**Pull request by jwilliams-ocient**
### SUMMARY

This PR addresses a bug that sometimes causes the Ocient engine to emit invalid polygon and linestring objects. See commit messages for more info.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/ocient.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
+import json
```
Comment (jwilliams-ocient) [inline superset/db_engine_specs/ocient.py:18]:
```suggestion
```

File: superset/db_engine_specs/ocient.py [hunk 2]
```diff
@@ -84,39 +84,87 @@ def _to_hex(data: bytes) -> str:
...
-    :returns: JSON representation of the linestring
+    :param point: the pyocient._STPoint instance
+    :returns: the geoJSON encoding of this point
     """
-    return f"{str([[p.long, p.lat] for p in linestring.points])}"
+    wkt_point = str(point)
+    return _wkt_to_geoJSON(wkt_point)
+
+
+def _linestring_to_geoJSON(linestring: Any) -> Any:
```
Comment (daveboutcher) [inline superset/db_engine_specs/ocient.py:123]: Why `Any`?  doesn't this take an STLinesTring and produce a known type?

**Review by daveboutcher — APPROVED** [review]

File: superset/db_engine_specs/ocient.py (hunk 2 above)
Comment (jwilliams-ocient) [reply superset/db_engine_specs/ocient.py]: resolved by using [type comments](https://mypy.readthedocs.io/en/stable/runtime_troubles.html#string-literal-types-and-type-comments)

**Comment by jwilliams-ocient** [comment]
Closing and reopening to kick off new CI run

**Comment by jwilliams-ocient** [comment]
There's a CSV test broken on the master branch. Once the fix has been merged I will close/reopen

**Review by rusackas — APPROVED** [review]
LGTM, but would love to get some additional eyes on this, so pinging @villebro @john-bodley  @michael-s-molina for extra 👀

**Review by villebro — APPROVED** [review]
Looks great, thanks for the added tests!

**Comment by jwilliams-ocient** [comment]
Alright, I think this one is ready to merge. Thanks @villebro , @rusackas !

**Comment by rusackas** [comment]
Thank YOU!

**Pull request merged**